src/types/jsx.ts

```diff
@@ -5,1126 +5,1125 @@ import { CSS } from "../css"
-      extends Omit<SelectHTMLAttributes<T>, "onChange" | "onInput"> {
-      multiple?: false
-      onChange?: (event: ChangeEvent<string>) => void
-      onInput?: (event: InputEvent<string>) => void
-    }
-
-    export interface SelectHTMLAttributes<T> extends HTMLAttributes<T> {
-      autoFocus?: boolean
-      autoComplete?: string
-      autocomplete?: string
-      disabled?: boolean
-      form?: string
-      multiple?: boolean
-      name?: string
-      required?: boolean
-      size?: number
-    }
-
-    export interface SourceHTMLAttributes<T> extends HTMLAttributes<T> {
-      media?: string
-      sizes?: string
-      src?: string
-      srcSet?: string
-      type?: string
-    }
-
-    export interface StyleHTMLAttributes<T> extends HTMLAttributes<T> {
-      media?: string
-      nonce?: string
-      scoped?: boolean
-      type?: string
-    }
-
-    export interface TableHTMLAttributes<T> extends HTMLAttributes<T> {
-      cellPadding?: number | string
-      cellpadding?: number | string
-      cellSpacing?: number | string
-      cellspacing?: number | string
-      summary?: string
-    }
-
-    export interface TextareaHTMLAttributes<T> extends HTMLAttributes<T> {
-      // Purview specific
-      defaultValue?: string | number
-
-      // Standard HTML attributes
-      autoFocus?: boolean
-      autofocus?: boolean | string
-      cols?: number
-      disabled?: boolean
-      form?: string
-      maxLength?: number
-      maxlength?: number | string
-      minLength?: number
-      minlength?: number | string
-      name?: string
-      placeholder?: string
-      readOnly?: boolean
-      readonly?: boolean | string
-      required?: boolean
-      rows?: number
-      value?: string | number
-      wrap?: string
-    }
-
-    export interface TdHTMLAttributes<T> extends HTMLAttributes<T> {
-      colSpan?: number
-      colspan?: number
-      headers?: string
-      rowSpan?: number
-    }
-
-    export interface ThHTMLAttributes<T> extends HTMLAttributes<T> {
-      colSpan?: number
-      colspan?: number
-      headers?: string
-      rowSpan?: number
-      rowspan?: number | string
-      scope?: string
-    }
-
-    export interface TimeHTMLAttributes<T> extends HTMLAttributes<T> {
-      dateTime?: string
-    }
-
-    export interface TrackHTMLAttributes<T> extends HTMLAttributes<T> {
-      default?: boolean
-      kind?: string
-      label?: string
-      src?: string
-      srcLang?: string
-      srclang?: string
-    }
-
-    export interface VideoHTMLAttributes<T> extends MediaHTMLAttributes<T> {
-      height?: number | string
-      playsInline?: boolean
-      playsinline?: boolean | string
-      poster?: string
-      width?: number | string
-    }
-
-    export type Child =
-      | string
-      | number
-      | boolean
-      | null
-      | undefined
-      | JSX.Element
-
-    export interface HTMLAttributes<T = HTMLElement> extends DOMAttributes {
```

Comment:
See here that the generic type T is unused.

Reply:
In fact, Preact does use a generic type like you have here, but that is because its subtypes depend on the type of the element being passed as a type argument, which represents the `EventTarget` for for that HTML element's call handlers (see the screenshot below). 

Purview's event handlers do not contain the concept of an Event Target, so Preact's implementation of the `HTMLAttributes` interface is semantically different than Purview's. 

![image](https://github.com/karthikv/purview/assets/129202431/f145f5aa-c53d-4bf5-9f29-9f6ccbf9db9e)

I wasn't easily able to find other JSX types. I suppose that finding open source JSX definitions is not as necessary as it was when you created Purview?  If you would like to chat about this please let me know. 

From Preact's code for your reference:

The `RefType` generic is passed to the extended interfaces `ClassAttributes<T>` and `DOMAttributes<Target extends EventTarget>`. It is not used within `HTMLAttributes`' own interface definition.
```
export interface HTMLAttributes<RefType extends EventTarget = EventTarget>
	extends ClassAttributes<RefType>,
		DOMAttributes<RefType>,
		AriaAttributes {
	// Standard HTML Attributes
	accept?: string | undefined | SignalLike<string | undefined>;
	acceptCharset?: string | undefined | SignalLike<string | undefined>;
	..._extensive attributes list, no use of RefType_...
}

```
No particular use of the RefType above inside `ClassAttributes` except that `Ref<T>` must take a type.
```
export interface ClassAttributes<T> extends Attributes {
	ref?: Ref<T>;
}
```

The type argument taken by `Ref<T>` is in fact unconstrained, so this is not an important part of the code semantically. See here the relevant type definitions for `Ref<T>`:

```
export type RefObject<T> = { current: T | null };
export type RefCallback<T> = (instance: T | null) => void;
export type Ref<T> = RefObject<T> | RefCallback<T> | null;
```

Here's where the `RefType` from `HTMLAttributes` actually becomes important:
```
	export interface DOMAttributes<Target extends EventTarget>
		extends PreactDOMAttributes {
		// Image Events
		onLoad?: GenericEventHandler<Target> | undefined;
		onLoadCapture?: GenericEventHandler<Target> | undefined;
		onError?: GenericEventHandler<Target> | undefined;
		onErrorCapture?: GenericEventHandler<Target> | undefined;
                ..._comprehensive event handler method types_...
	}

```

However as you can see, that is because the various event handlers require a `Target` type argument.

Reply:
Sounds good, thanks!